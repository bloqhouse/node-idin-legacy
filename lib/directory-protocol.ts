import { pd } from 'pretty-data'
import { MERCHANT_ID, MERCHANT_SUBID} from './constants'
import { signXML } from './idin-protocol'

export function formatDirectoryProtocolXML() {
  const xml = pd.xmlmin(`
    <?xml version="1.0" encoding="UTF-8"?>
    <DirectoryReq version="1.0.0" productID="NL:BVN:BankID:1.0" xmlns="
    http://www.betaalvereniging.nl/iDx/messages/Merchant-Acquirer/1.0.0"
    xmlns:ds="http://www.w3.org/2000/09/xmldsig#" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
      <createDateTimestamp>${new Date().toISOString()}</createDateTimestamp>
      <Merchant>
        <merchantID>${MERCHANT_ID}</merchantID>
        <subID>${MERCHANT_SUBID}</subID>
      </Merchant>
    </DirectoryReq>
  `)

  return signXML(xml)
}
