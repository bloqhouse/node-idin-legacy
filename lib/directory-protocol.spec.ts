import { formatDirectoryProtocolXML } from './directory-protocol'

describe('format directory protocol', () => {
  it('it works correctly', () => {
    expect(formatDirectoryProtocolXML()).toMatchSnapshot()
  })
})
