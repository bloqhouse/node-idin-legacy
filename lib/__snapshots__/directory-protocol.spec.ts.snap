// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`format directory protocol it works correctly 1`] = `"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?><DirectoryReq version=\\"1.0.0\\" productID=\\"NL:BVN:BankID:1.0\\" xmlns=\\"http://www.betaalvereniging.nl/iDx/messages/Merchant-Acquirer/1.0.0\\" xmlns:ds=\\"http://www.w3.org/2000/09/xmldsig#\\" xmlns:xsi=\\"http://www.w3.org/2001/XMLSchema-instance\\"><createDateTimestamp>2017-10-09T20:08:32.110Z</createDateTimestamp><Merchant><merchantID>0030001645</merchantID><subID>0</subID></Merchant><Signature xmlns=\\"http://www.w3.org/2000/09/xmldsig#\\"><SignedInfo><CanonicalizationMethod Algorithm=\\"http://www.w3.org/2001/10/xml-exc-c14n#\\"/><SignatureMethod Algorithm=\\"http://www.w3.org/2001/04/xmldsig-more#rsa-sha256\\"/><Reference URI=\\"\\"><Transforms><Transform Algorithm=\\"http://www.w3.org/2000/09/xmldsig#enveloped-signature\\"/><Transform Algorithm=\\"http://www.w3.org/2001/10/xml-exc-c14n#\\"/></Transforms><DigestMethod Algorithm=\\"http://www.w3.org/2001/04/xmlenc#sha256\\"/><DigestValue>zUcZHEZDLEui0dRUSkOVxpH9zB/bdnLt1V0/O4SkeOo=</DigestValue></Reference></SignedInfo><SignatureValue>gq/DHTTl48dmGLiYm1jiIb7R5OZwahdBoarWHcW+d6u1D+Pb8ZyGTAv3iAb8HjH0EIlMe/rWICubNnW2uZwh3DPpzpJ9C0818oYPzArqyRg3drsh8hVPUi+eumG24m7j21yMJ6KIRm66TxISa89h3bMf262tQmcI39DE6JD7tR0rGmmo19dJlaa9LOfstNjqClwoS1cBvhCa2szniOMP3OuRyHCi2AS9KQZVKXXzRuhkmHMuAn86xYwCfzPWS/QwqtYHUXzhaC7BNKCZXNxh02t72RAQuYrUKFtebx9+3lw66MsFpQKc0QLt1h+qobpLhURmb9Bs7II2wefK0hMleg==</SignatureValue><KeyInfo><KeyName>8c6415e72e553d2a9bda7afc9fcc0a5ec5d82841</KeyName></KeyInfo></Signature></DirectoryReq>"`;
